IPython/nbconvert/filters/markdown_mistune.py

```diff
@@ -23,7 +23,7 @@
 
 @undoc
 class MathBlockGrammar(mistune.BlockGrammar):
-    block_math = re.compile("^\$\$(.*?)\$\$", re.DOTALL)
+    block_math = re.compile("^\$\$([^\$]*?)\$\$", re.DOTALL)
```

Comment:
I don't understand how these regex changes can make a difference - given that we were using non-greedy patterns before, shouldn't this match the same thing? I trust you that it does fix it, I'd just like to understand better.